journal/202208.md

```diff
@@ -0,0 +1,571 @@
+- Censorship reason and moderator is shown for censored proposals.
+- Improved drafts UX.
+- Initial work for the new architecture for building Politeia-like apps.
+
+See full release notes in [politeia](https://github.com/decred/politeia/releases/tag/v1.4.0) and [politeiagui](https://github.com/decred/politeiagui/releases/tag/v1.4.0) repositories.
+
+
+## Development
+
+The work reported below has the "merged to master" status unless noted otherwise. It means that the work is completed, reviewed, and integrated into the source code that advanced users can [build and run](https://medium.com/@artikozel/the-decred-node-back-to-the-source-part-one-27d4576e7e1c), but is not yet available in release binaries for regular users.
+
+
+### dcrd
+
+_[dcrd](https://github.com/decred/dcrd) is a full node implementation that powers Decred's peer-to-peer network around the world._
+
+dcrd v1.7.4 was released to fix testnet mining algorithm as detailed below. It was a source code only release as it is primarily intended for developers { right? }.
+
+Merged in `master` and backported to v1.7.4 release:
+
+- Enforced [difficulty throttling](https://github.com/decred/dcrd/pull/2978) on testnet. Normally ASICs are not expected on testnet and blocks are mined with "slow" CPU mining, since it's not reasonable to require high-powered hardware to run tesnet. There is no financial incentive either because tesnet coins have no value and ASIC miners have a better choice to mine mainnet coins for profit. Despite that, a bunch of ASICs joined Decred tesnet and ramped up the difficulty, which effectively stalled the network. To limit thea type of games that ASICs can play on testnet two new rules are introduced: a limit on maximum allowed difficulty and a limit on block production rate once maximum difficulty is reached. This should keep CPU mining feasible even in the presence of GPUs and ASICs. It should be noted that this solution is only suitable on a test network where no real monetary value is in play and thus the typical game theory mechanics do not apply. Finally, this change will invalidate the high-work chain which has stalled testnet.
+  - { _Do we know how many Th/s or ASIC units were used to stall testnet? Any other details to entertain the reader with?_ }
+
+- Modified handling of blocks with too few voting tickets to also [copy the regular transactions](https://github.com/decred/dcrd/pull/2978/commits/659b7fed1f225861b8b689ead1a0466099992479) into an alternative block template in addition to the stake transactions. This makes template creation more robust in the case of duplicate spends and helps ensure that all transactions in an invalidated (under-voted) block are added to the alternative block templates.
+  - { _It was part of #2978 but I felt it's fun to write about. If it's minor, we can just drop it. If keeping, how did it work without copying regular tx before? Did those tx make it into alternative blocks in some other way?_ }
```

Comment:
Yes, they would just make it into the next block.  Effectively, just an unnecessary delay.